tests/acceptance/bind-data-test-attributes-in-components-test.js

```diff
@@ -0,0 +1,51 @@
+import { test } from 'qunit';
+import moduleForAcceptance from '../../tests/helpers/module-for-acceptance';
+
+import config from 'dummy/config/environment';
+
+if (!config.stripTestSelectors) {
+
+  /*
+   * We use an acceptance test here to test the "ember-test-selectors" initializer,
+   * because initializers are only applied in acceptance tests, but not in
+   * component integration tests.
+   */
+  moduleForAcceptance('Acceptance | Initializer | ember-test-selectors', {
+    beforeEach() {
+      visit('/bind-test');
+    },
+  });
+
+  test('it binds data-test-* attributes on components', function (assert) {
+    assert.equal(find('.test1').find('div[data-test-first]').length, 1, 'data-test-first exists');
+    assert.equal(find('.test1').find('div[data-test-first="foobar"]').length, 1, 'data-test-first has correct value');
+  });
+
+  test('it binds data-test-* attributes on components in block form', function (assert) {
+    assert.equal(find('.test2').find('div[data-test-first]').length, 1, 'data-test-first exists');
+    assert.equal(find('.test2').find('div[data-test-first="foobar"]').length, 1, 'data-test-first has correct value');
+  });
+
+  test('it works with multiple data-test-* attributes on components', function (assert) {
+    assert.equal(find('.test3').find('div[data-test-first]').length, 1, 'data-test-first exists');
+    assert.equal(find('.test3').find('div[data-test-first="foobar"]').length, 1, 'data-test-first has correct value');
+    assert.equal(find('.test3').find('div[data-test-second]').length, 1, 'data-test-second exists');
+    assert.equal(find('.test3').find('div[data-test-second="second"]').length, 1, 'data-test-second has correct value');
+  });
+
+  test('it leaves other data attributes untouched, when a data-test-* attribute is present as well on components', function (assert) {
+    assert.equal(find('.test4').find('div[data-test-first]').length, 1, 'data-test-first exists');
+    assert.equal(find('.test4').find('div[data-test-first="foobar"]').length, 1, 'data-test-first has correct value');
+    assert.equal(find('.test4').find('div[data-non-test]').length, 0, 'data-non-test does not exists');
+  });
+
+  test('it leaves data-test attributes untouched on components', function (assert) {
+    assert.equal(find('.test5').find('div[data-test]').length, 0, 'data-test does not exists');
+  });
+
+  test('it leaves other data attributes untouched on components', function (assert) {
+    assert.equal(find('.test6').find('div[data-non-test]').length, 0, 'data-non-test does not exists');
+  });
+
+}
```

Comment:
Since `attributeBindings` is a concatenated property, the string will end up in an array eventually. This only works for:

```js
attributeBindings: "myAttribute"
```

which will become

`attributeBindings: ["myAttribute"]`

and will not work for:

```js
attributeBindings: "myAttribute myOtherAttribute"
```

which would result in `attributeBindings: ["myAttribute myOtherAttribute"]`